src/ai/backend/agent/server.py

```diff
@@ -162,15 +164,18 @@ async def _inner(self: AgentRPCServer, *args, **kwargs):
 
 class RPCFunctionRegistry:
     functions: Set[str]
+    _metric_observer: RPCMetricObserver
 
     def __init__(self) -> None:
         self.functions = set()
+        self._metric_observer = RPCMetricObserver.instance()
```

Comment:
Is it used as a singleton object? Is it expected to use this observer in other context?

Reply:
This code was added to handle a case where two or more gauges are initialized, which causes panic in prometheus_client.